miso-web/src/main/webapp/scripts/hot.js

```diff
@@ -929,33 +929,79 @@ var HotUtils = {
       }
     }, ];
   },
-  makeParents: function(slug, parentCategories) {
+  
+  makeParents: function(slug, parentCategories){
+  	return HotUtils.makeRelations(slug + '/parents', 'Parents', parentCategories, true);
+  },
+  
+  makeChildren: function(slug, childCategories){
+  	return HotUtils.makeRelations(slug + '/children', 'Children', childCategories, false);
+  },
+  
+  makeRelations: function(slug, relationship, relationCategories, useParentBound) {
 
     return {
-      name: "Parents",
+      name: relationship,
       action: function(items) {
-        Utils.showWizardDialog('Parents', parentCategories.map(function(category) {
+      	function makeCategoriesFilter(items){
+      		if(!Constants.isDetailedSample){
+      			return function(category){
+      				if(useParentBound){
+      					return category.index >= Constants.sampleCategories.length;
+      				}
+      				return category.index > Constants.sampleCategories.length;
+      			};
+      		}
+      		var childBound = Constants.sampleCategories.length - 1;
+      		var parentBound = 0;
+      		for(sample in items){
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Add the "let", "const" or "var" keyword to this declaration of "sample" to make it explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2703)